apps/dashboard/app/models/script.rb

```diff
@@ -61,30 +61,44 @@ def initialize(opts = {})
     @created_at = opts[:created_at]
     sm_opts = {
       form:       opts[:form] || [],
-      attributes: opts[:attributes] || {}
+      attributes: opts[:attributes] || {},
+      job_environment: opts[:job_environment] || {},
     }
 
     add_required_fields(**sm_opts)
+
     @smart_attributes = build_smart_attributes(**sm_opts)
   end
 
-  def build_smart_attributes(form: [], attributes: {})
-    form.map do |form_item_id|
+  def build_smart_attributes(form: [], attributes: {}, job_environment: {})
+    attrs = form.map do |form_item_id|
```

Comment:
I don't think we need the addition of `job_environment` here - it should work just like all the other smart attributes.

I think the structure of this is a bit off - like you're doing a lot of the work in this model when a lot of this parsing & updating logic should be embedded in the smart attribute itself.

I.e., it knows how to build itself and apply attributes to it.

So given a YAML like this:
```yaml
form:
  - auto_environment_variable_FOO
attributes:
  auto_environment_variable_FOO:
    value: bar
```

That would call the factory method like 

```ruby
#  SmartAttributes::AttributeFactory.build(form_item_id, attrs) evaluates to 
SmartAttributes::AttributeFactory.build('auto_environment_variable_FOO', { value: 'bar' })
```

Then in the factory - we can do something similar to what `auto_modules` does - the factory parses out FOO and supplies it as an argument to the builder 

https://github.com/OSC/ondemand/blob/a61aef6fc2cf145d0903189b7b0f2a977c6bd40c/apps/dashboard/app/lib/smart_attributes/attribute_factory.rb#L13-L17